pkg/vz/vz_driver_darwin.go

```diff
@@ -204,6 +207,34 @@ func (l *LimaVzDriver) RunGUI() error {
 	return fmt.Errorf("RunGUI is not supported for the given driver '%s' and display '%s'", "vz", *l.Instance.Config.Video.Display)
 }
 
+func (l *LimaVzDriver) RuntimeConfig(_ context.Context, config interface{}) (interface{}, error) {
+	if config == nil {
+		return l.config, nil
+	}
+	var newConfig LimaVzDriverRuntimeConfig
+	err := mapstructure.Decode(config, &newConfig)
+	if err != nil {
+		return nil, err
+	}
+	if newConfig.SaveOnStop {
+		if runtime.GOARCH != "arm64" {
+			return nil, fmt.Errorf("saveOnStop is not supported on %s", runtime.GOARCH)
+		} else if runtime.GOOS != "darwin" {
+			return nil, fmt.Errorf("saveOnStop is not supported on %s", runtime.GOOS)
+		} else if macOSProductVersion, err := osutil.ProductVersion(); err != nil {
+			return nil, fmt.Errorf("failed to get macOS product version: %w", err)
+		} else if macOSProductVersion.Major < 14 {
+			return nil, fmt.Errorf("saveOnStop is not supported on macOS %d", macOSProductVersion.Major)
+		}
```

Comment:
This looks like a copy of the block validating the config in a previous commit. Maybe we can have one place doing this check:

    save_restore.Validate(instance) error

Then we can call it in both places,